rust_src/src/vectors.rs

```diff
@@ -346,20 +337,21 @@ impl LispBoolVecRef {
         unsafe { self.set_unchecked(idx, b) }
     }
 
-    pub fn set_checked(&mut self, idx: isize, b: bool) {
-        if idx < 0 || idx >= self.len() as isize {
+    pub fn set_checked(&mut self, idx: usize, b: bool) {
+        if idx >= self.len() {
             args_out_of_range!(self.as_lisp_obj(), LispObject::from(idx));
         }
 
-        unsafe { self.set_unchecked(idx as usize, b) }
+        unsafe { self.set_unchecked(idx, b) }
     }
 
     pub unsafe fn set_unchecked(&mut self, idx: usize, b: bool) {
-        let limbp = self.as_mut_byte_ptr().offset(idx as isize / 8);
+        let limb = self.as_mut_slice()
+            .get_unchecked_mut(idx / BITS_PER_BITS_WORD as usize) as *mut usize;
         if b {
-            *limbp |= 1 << (idx % 8)
+            *limb |= 1 << (idx % BITS_PER_BITS_WORD as usize)
```

Comment:
At some point we might want to remove our usage of BITS_PER_BITS_WORD. I don't think Remacs (or even GNU Emacs) will run on a machine where a byte isn't 8 bits.

Reply:
BITS_PER_BITS_WORD is actually 64; a `bits_word` is generally a tyepdef for a `long`.